ols/src/query_helpers/query_docs.py

```diff
@@ -0,0 +1,124 @@
+                    Default: 20)
+                    lambda_mult: Diversity of results returned by MMR;
+                        1 for minimum diversity and 0 for maximum. (Default:
+                        0.5)
+                    filter: Filter by document metadata
+
+        Returns:
+            List[Document]: Returns a list of docs.
+
+        Examples:
+        .. code-block:: python
+
+            # Retrieve more documents with higher diversity
+            # Useful if your dataset has many similar documents
+            docsearch.as_retriever(
+                search_type="mmr",
+                search_kwargs={'k': 6, 'lambda_mult': 0.25}
+            )
+
+            # Fetch more documents for the MMR algorithm to consider
+            # But only return the top 5
+            docsearch.as_retriever(
+                search_type="mmr",
+                search_kwargs={'k': 5, 'fetch_k': 50}
+            )
+
+            # Only retrieve documents that have a relevance score
+            # Above a certain threshold
+            docsearch.as_retriever(
+                search_type="similarity_score_threshold",
+                search_kwargs={'score_threshold': 0.8}
+            )
+
+            # Only get the single most similar document from the dataset
+            docsearch.as_retriever(search_kwargs={'k': 1})
+
+            # Use a filter to only retrieve documents from a specific paper
+            docsearch.as_retriever(
+                search_kwargs={'filter': {'paper_title':'GPT-4 Technical
+                Report'}}
+            )
+        """
+        self.logger.info(
+            f"""Retrieving docs for
+                query: {query},
+                perform_query: {perform_query},
+                vectordb: {vectordb},
+                search_type: {search_type},
+                search_kwargs: {search_kwargs}
+        """
+        )
+
+        if not perform_query:
+            return None
+
+        if not (
+            search_type == "mmr"
+            or search_type == "similarity"
+            or search_type == "similarity_score_threshold"
+        ):
+            self.logger.error(f"incorrect search type {search_type}")
+            raise RetrieveDocsExceptionError("search type in invalid")
+
+        db_retriever: VectorStoreRetriever = vectordb.as_retriever(
+            search_type=search_type, search_kwargs=search_kwargs["search_kwargs"]
+        )
+
+        try:
+            docs = db_retriever.get_relevant_documents(query=query)
+        except Exception:
+            self.logger.error(
+                f"exception raised while getting the docs for query: {query}"
+            )
+            raise RetrieveDocsExceptionError(
+                "error in getting the  docs from vectorstore"
+            )
+
+        return docs
+
+
+class RetrieveDocsExceptionError(Exception):
+    """Raise this exception when there is an issue in retrieving the docs."""
```

Comment:
The [Google docstring standard](https://google.github.io/styleguide/pyguide.html#384-classes) (we use) implies a specific syntactic of the exception docstring. We are [already using](https://github.com/openshift/lightspeed-service/blob/main/ols/src/llms/llm_loader.py#L26) it. So this should be something like "Issue with retrieving the docs."

Reply:
done